pyincore/client.py

```diff
@@ -410,23 +429,144 @@ def delete(self, url: str, timeout=(30, 600), **kwargs):
 
         return return_http_response(r)
 
-    def create_service_json_entry(self):
-        update_hash_entry(
-            "add", hashed_url=self.hashed_service_url, service_url=self.service_url
-        )
+    def clear_cache(self):
+        """
+        This function helps clear the data cache for a specific repository or the entire cache
 
-    @staticmethod
-    def clear_root_cache():
+        Returns: None
+
+        """
         # incase cache_data folder doesn't exist
         if not os.path.isdir(pyglobals.PYINCORE_USER_DATA_CACHE):
             logger.warning("User data cache does not exist")
             return None
 
-        for root, dirs, files in os.walk(pyglobals.PYINCORE_USER_DATA_CACHE):
-            for d in dirs:
-                shutil.rmtree(os.path.join(root, d))
-        update_hash_entry("clear")
-        return None
+        if not os.path.isdir(self.hashed_svc_data_dir):
+            logger.warning("Cached folder doesn't exist")
+            return None
+        try:
+            shutil.rmtree(self.hashed_svc_data_dir)
+        except PermissionError as e:
+            print(f"Error clearing cache : {e}")
+
+        # clear entry from service.json
+        update_hash_entry("edit", hashed_url=self.hashed_service_url)
+        return
+
+
+class IncoreInternalClient(Client):
+    """IN-CORE service client class meant for interal communiation with restful services behind the auth middleware
```

Comment:
communication

Reply:
I modified the code to combine internal client with the current incore client. Those docstrings should be no longer applicable.